lib/presentation/widget/myspace/my_space_widget.dart

```diff
@@ -160,27 +162,45 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
                     })
                 ],
               ),
-              floatingActionButton: StoreConnector<AppState, SelectMode>(
-                converter: (store) => store.state.mySpaceState.selectMode,
-                builder: (context, selectMode) {
-                  return selectMode == SelectMode.INACTIVE ? FloatingActionButton(
-                    key: Key('my_space_upload_button'),
-                    onPressed: () => mySpaceViewModel
-                        .openUploadFileMenu(context, _uploadFileMenuActionTiles(context)),
-                    backgroundColor: AppColor.primaryColor,
-                    child: SvgPicture.asset(
-                      imagePath.icPlus,
-                      width: 24,
-                      height: 24,
-                    ),
-                  ) : SizedBox.shrink();
+          bottomNavigationBar: StoreConnector<AppState, AppState>(
+                  converter: (store) => store.state,
+                  builder: (context, appState) {
+                    if (!_isInSearchState(appState.uiState) &&
+                        appState.mySpaceState.selectMode == SelectMode.INACTIVE) {
+                      return SearchBottomBarBuilder()
+                          .key(Key('search_bottom_bar'))
+                          .onSearchActionClick(() => mySpaceViewModel.openSearchState())
+                          .build();
+                    }
+                    return SizedBox.shrink();
+                  }),
+              floatingActionButton: StoreConnector<AppState, AppState>(
+                converter: (store) => store.state,
+                builder: (context, appState) {
+                  if (!_isInSearchState(appState.uiState) &&
+                      appState.mySpaceState.selectMode == SelectMode.INACTIVE) {
+                    return FloatingActionButton(
+                      key: Key('my_space_upload_button'),
+                      onPressed: () => mySpaceViewModel.openUploadFileMenu(context, _uploadFileMenuActionTiles(context)),
+                      backgroundColor: AppColor.primaryColor,
+                      child: SvgPicture.asset(
+                        imagePath.icPlus,
+                        width: 24,
+                        height: 24,
+                      ),
+                    );
+                  }
+                  return SizedBox.shrink();
                 }),
-              floatingActionButtonLocation:
-                  FloatingActionButtonLocation.centerFloat,
+              floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
             ),
         converter: (Store<AppState> store) => mySpaceViewModel);
   }
 
+  bool _isInSearchState(UIState uiState) {
```

Comment:
This is why I've made selectors before, I think we need to put it in a ui_selector file so we can use it in the whole app

Reply:
cool, man

Reply:
but I'll put it inside `UIState` class as an extension. Sounds good to you?

Reply:
Sure!